src/store/generator/hooks/useHighlightRequestChanges.ts

```diff
@@ -0,0 +1,119 @@
+import { diffWords } from 'diff'
+import { useMemo } from 'react'
+
+import { KeyValueTuple, ProxyDataWithMatches } from '@/types'
+import { Match } from '@/types/fuse'
+import { diffChangesToFuseIndices } from '@/utils/diff'
+
+import { selectFilteredRequests, useGeneratorStore } from '..'
+
+export function useHighlightRequestChanges(
+  requests: ProxyDataWithMatches[]
+): ProxyDataWithMatches[] {
+  const originalRequests = useGeneratorStore(selectFilteredRequests)
+
+  return useMemo(() => {
+    return requests.map((data) => {
+      const originalRequest = originalRequests.find(
+        (request) => request.id === data.id
+      )?.request
+
+      if (!originalRequest) {
+        return data
+      }
+
+      return addHighlights(originalRequest, data)
+    })
+  }, [requests, originalRequests])
+}
+
+function addHighlights(
+  originalRequest: ProxyDataWithMatches['request'],
+  data: ProxyDataWithMatches
+) {
+  // Don't overwrite search matches when present
```

Comment:
I find this behavior confusing, because it doesn't match how other tools work.
I don't think this should be a blocking comment as long as we can create a ticket (and add it to stretch goals so it doesn't get lost) to implement more flexible text decorations.

Reply:
This one is a bit tricky, since we are using same methods of highlighting for both search and diff, we can't show both highlights when the same string is matched. I've checked how diff editor handles this, and it works similar to this - search match overwrites diff highlight:

No search, `admin` highlighted in green
![screencapture 2025-05-05 at 14 23 26](https://github.com/user-attachments/assets/adc491ef-35fd-4238-9e27-f8979dd8fbda)

Search match - `admin` becomes highlighted in gray
![screencapture 2025-05-05 at 14 23 34](https://github.com/user-attachments/assets/a7f5a7e4-f852-4d7c-8cee-cc2a20873595)

The only difference is that monaco still highlights the whole line as an "addition", is that what you have in mind?

Reply:
No, I was thinking about supporting multiple kinds of text nodes/decorations, that can overlap each other. For example, Google Docs doesn't overwrite search matches when they overlap with the comment highlighting:

<img width="712" alt="image" src="https://github.com/user-attachments/assets/40d809b4-ed34-4ace-ab1d-7b6a0a3af58b" />